src/Apps/Sell/Routes/ArtistRoute.tsx

```diff
@@ -0,0 +1,178 @@
+import { Spacer, Text } from "@artsy/palette"
+import {
+  ArtistAutoComplete,
+  AutocompleteArtist,
+} from "Apps/Consign/Routes/SubmissionFlow/ArtworkDetails/Components/ArtistAutocomplete"
+import { DevDebug } from "Apps/Sell/Components/DevDebug"
+import { SubmissionLayout } from "Apps/Sell/Components/SubmissionLayout"
+import { useSellFlowContext } from "Apps/Sell/SellFlowContext"
+import { RouterLink } from "System/Router/RouterLink"
+import { useRouter } from "System/Router/useRouter"
+import {
+  ArtistRoute_submission$data,
+  ArtistRoute_submission$key,
+} from "__generated__/ArtistRoute_submission.graphql"
+import { Formik, FormikProps } from "formik"
+import * as React from "react"
+import { graphql, useFragment } from "react-relay"
+import * as Yup from "yup"
+
+const FRAGMENT = graphql`
+  fragment ArtistRoute_submission on ConsignmentSubmission {
+    internalID
+    artist {
+      internalID
+      targetSupply {
+        isTargetSupply
+      }
+      name
+    }
+  }
+`
+
+const Schema = Yup.object().shape({
+  artistName: Yup.string().required(),
+  artistId: Yup.string().required(),
+  isTargetSupply: Yup.boolean().isTrue(),
+})
+
+interface FormValues {
+  artistId: string
+  artistName: string
+  isTargetSupply: boolean
+}
+
+interface ArtistRouteProps {
+  submission: ArtistRoute_submission$key
+}
+
+export const ArtistRouteFragmentContainer: React.FC<ArtistRouteProps> = props => {
+  const submission = useFragment(FRAGMENT, props.submission)
+
+  return <ArtistRoute submission={submission} />
+}
+
+export const ArtistRoute: React.FC<{
+  submission?: ArtistRoute_submission$data
+}> = ({ submission }) => {
+  const { router } = useRouter()
+  const { actions } = useSellFlowContext()
+
+  const isNewSubmission = !submission?.internalID
+
+  const onSubmit = async () => {}
+
+  const createSubmission = async (artist: AutocompleteArtist) => {
+    if (!artist?.internalID) return
+
+    const isTargetSupply = artist?.targetSupply?.isTargetSupply
+
+    if (!isTargetSupply) {
+      router.push(`/sell2/artist-not-eligible/${artist.internalID}`)
```

Comment:
I can't see this route being added - I think no need to push here since you are setting the formik store then rendering the required UI

Reply:
I do like having this as a separate screen though because it's well decoupled